### PR TITLE
 read rootcert from configured ns to connect to citadel

### PR DESCRIPTION
**Pull request by quanjielin**
fix issue raised in https://discuss.istio.io/t/nodeagent-crashing/2827

### CODE REVIEW

File: security/pkg/nodeagent/caclient/client.go [hunk 1]
```diff
@@ -26,19 +26,20 @@ import (
...
-	ns           = "istio-system"
-
+	googleCAName  = "GoogleCA"
+	citadelName   = "Citadel"
+	vaultCAName   = "VaultCA"
 	retryInterval = time.Second * 2
 	maxRetries    = 100
 )
 
+var namespace = env.RegisterStringVar("ISTIO_META_CONFIG_NAMESPACE", "istio-system", "namespace istio control plane run in").Get()
```
Comment (howardjohn) [inline security/pkg/nodeagent/caclient/client.go]: `ISTIO_META_` variables are intended to be passed through to envoy metadata. I don't think it makes sense to use this here.

File: security/pkg/nodeagent/caclient/client.go [hunk 2]
```diff
@@ -56,7 +57,7 @@ func NewCAClient(endpoint, caProviderName string, tlsFlag bool, tlsRootCert []by
 		if err != nil {
 			return nil, fmt.Errorf("could not create k8s clientset: %v", err)
 		}
-		controller := configmap.NewController(ns, cs.CoreV1())
+		controller := configmap.NewController(namespace, cs.CoreV1())
```
Comment (howardjohn) [inline security/pkg/nodeagent/caclient/client.go]: Is everything we are reading from always in the same namespace as the nodeagent? We have the concept of the "rootNamespace" which has all the global istio configs. So for example `*.local` destination rule, default sidecar, etc all live there. Should this be reading from that or the nodeagent namespace?

Comment (quanjielin) [reply security/pkg/nodeagent/caclient/client.go]: should be nodeagent namesapce (citadel write root cert to configmap where it's deployed, nodeagent read from it, and use it to connect to citadel)

File: security/pkg/nodeagent/caclient/client.go (hunk 1 above)
Comment (quanjielin) [reply security/pkg/nodeagent/caclient/client.go]: updated

File: security/pkg/nodeagent/caclient/client.go [hunk 3]
```diff
@@ -26,19 +26,20 @@ import (
...
-	ns           = "istio-system"
-
+	googleCAName  = "GoogleCA"
+	citadelName   = "Citadel"
+	vaultCAName   = "VaultCA"
 	retryInterval = time.Second * 2
 	maxRetries    = 100
 )
 
+var namespace = env.RegisterStringVar("NAMESPACE", "istio-system", "namespace istio control plane run in").Get()
```
Comment (howardjohn) [inline security/pkg/nodeagent/caclient/client.go]: `namespace istio control plane run in` is not accurate, this is the namespace Nodeagent is installed in. Pilot, etc can be installed other places

**Comment by quanjielin** [comment]
cherrypick this PR to 1.2, per request from https://github.com/istio/istio/issues/15929

**Pull request merged**